client/src/services/socket.js

```diff
@@ -0,0 +1,65 @@
+import io from "socket.io-client";
+import { SOCKET_SESSION } from "../constants";
+import store from "../store";
+import {
+  setNewMessage,
+  removeOfflineUser,
+  addOnlineUser,
+} from "../store/conversations";
+
+const socket = io(window.location.origin, { autoConnect: false });
+
+export const getSocket = () => socket;
```

Comment:
since this function only returns the socket object, any reason wrap this in a function?